system/lib/wasmfs/file.h

```diff
@@ -140,9 +140,13 @@ class Directory : public File {
   // This value was also copied from the existing file system.
   size_t getSize() override { return 4096; }
 
+  // File backend associated with this directory.
+  int backendID;
```

Comment:
I don't see an overview of the "ID" part of the design either here or in the other `.h` files - I think that would be good to add. That is, why do we use IDs here, why are they ints, why not use pointers there, etc. (I am actually unsure as to the answers, so these are not rhetorical questions! :smile: )

Reply:
For the design, I originally thought that we would represent backends with some sort of backend descriptor id. Similar to the open file table, we would have the `backendTable` vector that stores pointers to the actual backends themselves. So the user should be aware of the "id" of the backend that they are trying to target for example. 

ID numbers or some other identifiers are needed to determine which backend to open a file in. Not sure if you meant using raw pointers as a unique ID, but I don't think we should be passing the backend pointers to user code. Incremental IDs that assigned similar to `fd` in the open file table makes sense to me since you can also add any number of backends that you need.